qiskit_metal/_gui/widgets/qlibrary_display/file_model_qlibrary.py

```diff
@@ -15,13 +15,94 @@
+        self.path_imgs = path_imgs
+        self.defaultFilename = str(path_imgs / "metal_logo.png")
+        pixmap = QPixmap(self.defaultFilename).scaled(
+            QSize(self.size, self.size), Qt.KeepAspectRatio,
+            Qt.SmoothTransformation)
+        self.imageCache[self.defaultFilename] = pixmap
+
+    def data(self,
+             index: QModelIndex,
+             role: int = Qt.DisplayRole) -> typing.Any:
+        """
+        Sets standard size hint for indexes and allows
+        Args:
+            index(QModelIndex): Model Index holding data
+            role(int): DisplayRole being requested of index
+        Returns:
+            Any (QVariant): Data stored under the given role for the item referred to by the index.
+        """
+
+        # There are multiple columns so we don't want display an
+        # image in each one, (see filterAcceptsColumn)
+        if index.column() == 0:
+            if not self.isDir(index):
+                qfileinfo = self.fileInfo(index)
+                absoluteFilename = str(qfileinfo.absoluteFilePath())
+                if role == Qt.DecorationRole:
+                    matches = findProperty(absoluteFilename,
+                                           "\.\. image::[\r\n]+([^\r\n]+)")
+                    if matches is not None and len(matches) != 0:
+                        iconfile = matches[0].lstrip()
+                        pathFilename = self.path_imgs / "components" / iconfile
+                        if pathFilename.is_file():
+                            stringFilename = str(pathFilename)
+                            # imageCache pixmaps as they are seen
+                            if stringFilename in self.imageCache:
+                                return self.imageCache[stringFilename]
+                            else:
+                                pixmap = QPixmap(stringFilename).scaled(
+                                    QSize(self.size, self.size),
+                                    Qt.KeepAspectRatio, Qt.SmoothTransformation)
+                                self.imageCache[stringFilename] = pixmap
+                                return pixmap
+                    # display default image
+                    return self.imageCache[self.defaultFilename]
+                elif role == Qt.DisplayRole:
+                    relativeFilename = qfileinfo.fileName()
+                    if relativeFilename in self.nameCache:
+                        return self.nameCache[relativeFilename]
+                    else:
+                        # readfile = open(absoluteFilename, 'r')
+                        # filetext = readfile.read()
+                        # readfile.close()
+                        # matches = re.findall(
+                        #     "\.\. displayName::[\r\n]+([^\r\n]+)", filetext)
+                        matches = findProperty(
+                            absoluteFilename,
+                            "\.\. displayName::[\r\n]+([^\r\n]+)")
```

Comment:
Hmm, I see, so display name is good, but i thik this may not work in the docstring. 
The docs wont understand as of now (unless we add it) the sphinx rst command 

See here: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/images.html 

If we want to go this route wed want too define probably a custom sphinx directive 
https://stackoverflow.com/questions/12527783/how-to-create-custom-directives-in-sphinx
or just not define it as a directive